src/portable/sunxi/dcd_sunxi_musb.c

```diff
@@ -416,6 +423,8 @@ static unsigned find_free_memory(uint_fast16_t size_in_log2_minus3)
   }
   print_block_list(free_blocks, num_blocks);
 
+  USBC_SelectActiveEp(backup_ep);
```

Comment:
Please let me know why this operation is needed.  I would like to know if there is a bug in `dcd_musb.c`

I had designed `dcd_msub.c` that EPIDX register must be written when accessing the registers for FIFO settings(TXFIFOSZ, RXFIFOSZ, TXFIFOADD, and RXFIFOADD). My understanding is EPIDX effects only these registers. So, there is no need to keep the EPIDX value for a long time. If the EPIDX value needs to save and restore, I missed something condition.

Reply:
Hi Koji,

In my design, I didn't use the addressed endpoint registers. EPIDX is used in all places. In `dcd_edpt_open`, after calling `find_free_memory`, if current EP index is not restored, the following adjustments to FIFO addr and size will go to the wrong endpoints. In your code however, you adjusted EPIDX after `find_free_memory`, so you are not affected here.

Reply:
Thank you for your reply.

Ah, I see, you moved the statement of assigning a value to EPIDX register to before calling find_free_memory.